contracts/eosio.system/src/eosio.system.cpp

```diff
@@ -500,6 +582,67 @@ namespace eosiosystem {
       return false;
    }
 
+   // -------------------------------------------------------------------------------------------
+   struct canon_name_t
+   {
+      uint64_t _value; // all significant characters are shifted to the least significant position
```

Comment:
I think I prefer to the string version of the implementation. It is cleaner and easier to understand. This is used in account creation, not a frequently as transactions. Performance is important but not critical.

Reply:
I like the `uint64_t` implementation better personally, so waiting for someone to be the tie breaker on this :-).